### PR TITLE
charmc/ampicc: pass through certain compiler options even without file to compile

### PR DESCRIPTION
**Pull request by matthiasdiener**
Helps with (e.g.) configure scripts that parse the output of `-v`, `--version`, `-print-search-dirs` to detect compiler configurations.

### CODE REVIEW

File: src/libs/ck-libs/ampi/ampiCC [hunk 1]
```diff
@@ -75,6 +80,14 @@ done
 
 eval processArgs "$@"
 
+# If there are no args except -v or --version specified, just run
+# charmc with -v or --version. Helps with configure scripts that
+# parse the output to detect compiler versions.
+if [[ -z $ARGS ]] && [[ -n $SHOW_COMPILER_VERSION ]]; then
```
Comment (evan-charmworks) [inline src/libs/ck-libs/ampi/ampiCC]: Should `$ARGS` and `$SHOW_COMPILER_VERSION` be quoted here and once below or is the quoting practice for bash's `[[` different from `[` and `test`?

Comment (matthiasdiener) [reply src/libs/ck-libs/ampi/ampiCC]: Yeah, with `[[` you don't need to quote variables anymore (see e.g. https://stackoverflow.com/questions/3869072/test-for-non-zero-length-string-in-bash-n-var-or-var). If you prefer, I can add quotes if it makes it more consistent with the rest of the script.

**Comment by matthiasdiener** [comment]
Don’t merge this yet please. I’m still working on a small improvement

File: src/libs/ck-libs/ampi/ampiCC (hunk 1 above)
Comment (matthiasdiener) [reply src/libs/ck-libs/ampi/ampiCC]: Added quotes.

**Comment by matthiasdiener** [comment]
Ready for review.

**Pull request merged**